src/Maker/MakeUser.php

```diff
@@ -143,14 +138,17 @@ public function generate(InputInterface $input, ConsoleStyle $io, Generator $gen
         // need to write changes early so we can modify the contents below
         $generator->writeChanges();
 
-        // @legacy @todo - refactor for better naming conventions BEFORE next release.
-        $useAttributesForDoctrineMapping = $userClassConfiguration->isEntity() && ($this->doctrineHelper->isDoctrineSupportingAttributes()) && $this->doctrineHelper->doesClassUsesAttributes($userClassNameDetails->getFullName());
+        $entityUsesAttributes = ($isEntity = $userClassConfiguration->isEntity()) && $this->doctrineHelper->doesClassUsesAttributes($userClassNameDetails->getFullName());
+
+        if ($isEntity && !$entityUsesAttributes) {
+            throw new \RuntimeException('MakeUser only supports attribute mapping with doctrine entities.');
+        }
 
         // B) Implement UserInterface
         $manipulator = new ClassSourceManipulator(
             sourceCode: $this->fileManager->getFileContents($classPath),
             overwrite: true,
-            useAttributesForDoctrineMapping: $useAttributesForDoctrineMapping
+            useAttributesForDoctrineMapping: $entityUsesAttributes
```

Comment:
For make user we can gen a user that _is not_ an entity - in which case we don't add any attributes to the generated class.. But if the user _is_ an entity, we only generate it using attributes, otherwise we complain. Is my thinking correct on this?